### PR TITLE
Refactor SSLCertificates into Translator Package

### PR DESCRIPTION
**Pull request by spencerhance**
/assign @rramkumar1

### CODE REVIEW

**Comment by spencerhance** [comment]
@rramkumar1 I did some tests on removing the pre-shared cert check.

Here is the warning before:

```
Normal   SslCertificateMissing  13m (x10 over 14m)    loadbalancer-controller  googleapi: Error 404: The resource 'projects/shance-gke-dev/global/sslCertificates/fake-cert' was not found, notFound
```

Here is the warning after:
```
Warning  Sync  22s (x5 over 45s)     loadbalancer-controller  Error syncing to GCP: error running load balancer syncing routine: loadbalancer a85h3141-default-demo-ing-ngjmospa does not exist: googleapi: Error 404: The resource 'projects/shance-gke-dev/global/sslCertificates/fake-cert' was not found, notFound
```

I think this is just as easy to understand so I will remove it from the PR and re-push.  Our documentation also doesn't specify anything about the SslCertificateMissing event either.

**Review by rramkumar1 — COMMENTED** [review]
looks good to me

File: pkg/loadbalancers/certificates.go [hunk 1]
```diff
@@ -23,24 +23,30 @@ import (
...
 
 func (l *L7) checkSSLCert() error {
+	isL7ILB := flags.F.EnableL7Ilb && utils.IsGCEL7ILBIngress(l.runtimeInfo.Ingress)
+	tr := translator.NewTranslator(isL7ILB, l.namer)
+	env := &translator.Env{Region: l.cloud.Region(), Project: l.cloud.ProjectID()}
+	translatorCerts := tr.ToCompositeSSLCertificates(env, l.runtimeInfo.TLSName, l.runtimeInfo.TLS, l.Versions().SslCertificate)
+
+	// Check pre-shared certs.  This is not technically necessary, but it is easier
+	// to not remove existing warning events
+	if err := l.checkPreSharedCerts(translatorCerts); err != nil {
```
[the remark this answers is not in the document]
Comment (spencerhance) [reply pkg/loadbalancers/certificates.go]: Okay just pushed!

**Pull request merged**